config/config.go

```diff
@@ -0,0 +1,100 @@
+package config
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+	"slices"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+//go:generate paramgen -output=paramgen_config.go Config
+type Config struct {
+	// ClientID is the client id from the salesforce app
+	ClientID string `json:"clientID" validate:"required"`
+
+	// ClientSecret is the client secret from the salesforce app
+	ClientSecret string `json:"clientSecret" validate:"required"`
+
+	// OAuthEndpoint is the OAuthEndpoint from the salesforce app
+	OAuthEndpoint string `json:"oauthEndpoint" validate:"required"`
+
+	// gRPC Pubsub Salesforce API address
+	PubsubAddress string `json:"pubsubAddress" default:"api.pubsub.salesforce.com:7443"`
+
+	// InsecureSkipVerify disables certificate validation
+	InsecureSkipVerify bool `json:"insecureSkipVerify" default:"false"`
+
+	// Number of retries allowed per read before the connector errors out
+	RetryCount uint `json:"retryCount" default:"10"`
+
+	// TopicName {WARN will be deprecated soon} the TopicName the source connector will subscribe to
```

Comment:
lets mark this as deprecated, see how it was done for postgres [for example](https://github.com/ConduitIO/conduit-connector-postgres/blob/fffd4868e5d06327becb445bbb7afaaf23e1b3b3/source/config.go#L56), and check how it's crossed in the readme [config table](https://github.com/ConduitIO/conduit-connector-postgres/blob/fffd4868e5d06327becb445bbb7afaaf23e1b3b3/source/config.go#L56) .